main.js

```diff
@@ -0,0 +1,112 @@
+document.getElementById('submit').addEventListener('click', function () {
+
+  let name = document.getElementById('name').value;
+  let text = document.getElementById('message').value;
+
+  if (!name || !text) {
+    alert('Please fill in both name and text.');
+    return;
+  }
+
+  let postsDiv = document.querySelector('.posts');
+  let newPostDiv = document.createElement('div');
+  newPostDiv.classList.add('post')
+
+  let commentsButton = document.createElement('a');
+  commentsButton.href = "#";
+  commentsButton.classList.add('btn', 'comments-btn');
+  commentsButton.textContent = 'Comments';
+  newPostDiv.append(commentsButton);
+
+  let removeButton = document.createElement('a');
+  removeButton.href = "#";
+  removeButton.classList.add('btn', 'remove-btn');
+  removeButton.textContent = 'Remove';
+  removeButton.addEventListener('click', function () {
+    postsDiv.removeChild(newPostDiv)
+  });
+  newPostDiv.append(removeButton);
+
+  let newPostTextP = document.createElement('p');
+  let newPostTextNode = document.createTextNode(text);
+  newPostTextP.appendChild(newPostTextNode);
+  
+  let newPostNameP = document.createElement('p');
+  let newPostNameNode = document.createTextNode('Posted by: ' + name);
+  newPostNameP.appendChild(newPostNameNode);
+  
+  newPostDiv.append(newPostTextP.textContent + " - " + newPostNameP.textContent);
```

Comment:
you are appending a string and not a dom element, 

`newPostDiv.append(newPostTextP, newPostNameP);`